src/hermes/commands/deposit/invenio.py

```diff
@@ -551,19 +551,39 @@ def _get_license_identifier(ctx: CodeMetaContext, license_api_url: str):
             "Licenses of type 'CreativeWork' are not supported."
         )
 
-    parsed_url = urlparse(license_url)
-    url_path = parsed_url.path.rstrip("/")
-    license_id = url_path.split("/")[-1]
-
+    # Fetch full list of licenses available... maybe we should cache this.
     response = requests.get(
-        f"{license_api_url}/{license_id}", headers={"User-Agent": hermes_user_agent}
+        f"{license_api_url}?size=1000", headers={"User-Agent": hermes_user_agent}
```

Comment:
I haven't looked at this PR yet. Just one quick thing: It might be worth it to check `f"{license_api_url}/{license_id.lower()}"` first. before downloading all the licenses. See here: https://zenodo.org/api/vocabularies/licenses/apache-2.0

Reply:
Good point.

Reply:
Added in [2364bc3](https://github.com/hermes-hmc/workflow/pull/222/commits/2364bc34804776226557f06fce619be52b5145b3)